src/map/lua/lua_baseentity.cpp

```diff
@@ -4402,7 +4402,17 @@ inline int32 CLuaBaseEntity::canUsePet(lua_State *L)
 {
     DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
 
-    lua_pushboolean(L, m_PBaseEntity->loc.zone->CanUseMisc(MISC_PET));
+    if (dynamic_cast<CCharEntity*>(m_PBaseEntity) == nullptr)
```

Comment:
I think it's better to just DSP_DEBUG_BREAK if it's not a char and static_cast, since mobs cannot call a pet ever anyways using the player pet functions